library/ftd_install.py

```diff
@@ -0,0 +1,265 @@
+      - Device gateway of management interface. 
+      - If not specified, the module tries to fetch the existing value via REST API.
+    required: false
+    type: string
+  device_netmask:
+    description:
+      - Device netmask of management interface. 
+      - If not specified, the module tries to fetch the existing value via REST API.
+    required: false
+    type: string
+  dns_server:
+    description:
+      - DNS IP address of management interface. 
+      - If not specified, the module tries to fetch the existing value via REST API.
+    required: false
+    type: string
+  console_ip:
+    description:
+      - IP address of a terminal server. 
+      - Used to set up an SSH connection with device's console port through the terminal server.
+    required: true
+    type: string
+  console_port:
+    description:
+      - Device's port on a terminal server. 
+    required: true
+    type: string
+  console_username:
+    description:
+      - Username to login on a terminal server. 
+    required: true
+    type: string
+  console_password:
+    description:
+      - Password to login on a terminal server. 
+    required: true
+    type: string
+  tftp_server:
+    description:
+      - IP address of TFTP server with ROMmon image.
+    required: true
+    type: string
+  rommon_file_location:
+    description:
+      - Path to the boot (ROMmon) image on TFTP server.
+    required: true
+    type: string
+  image_file_location:
+    description:
+      - Path to the FTD pkg image to be transferred via HTTP.
+    required: true
+    type: string
+  image_version:
+    description:
+      - Version of FTD image to be installed.
+      - Helps to compare target and current FTD versions to prevent unnecessary reinstalls.
+    required: false
+    type: string
+"""
+
+EXAMPLES = """
+  - name: Install image v6.3.0 on FTD 5516
+    ftd_install:
+      device_hostname: firepower
+      device_password: pass
+      device_ip: 192.168.0.1
+      device_netmask: 255.255.255.0
+      device_gateway: 192.168.0.254
+      dns_server: 8.8.8.8
+
+      console_ip: 10.89.0.0
+      console_port: 2004
+      console_username: console_user
+      console_password: console_pass
+
+      tftp_server: 10.89.0.11
+      rommon_file_location: 'installers/ftd-boot-9.10.1.3.lfbff'
+      image_file_location: 'https://10.89.0.11/installers/ftd-6.3.0-83.pkg'
+      image_version: 6.3.0-83
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+
+try:
+    from kick.device2.ftd5500x.actions.ftd5500x import Ftd5500x
+    HAS_KICK = True
+except ImportError:
+    HAS_KICK = False
+
+try:
+    from ansible.module_utils.configuration import BaseConfigurationResource, ParamName, PATH_PARAMS_FOR_DEFAULT_OBJ
+except ImportError:
+    from module_utils.configuration import BaseConfigurationResource, ParamName, PATH_PARAMS_FOR_DEFAULT_OBJ
+
+GET_SYSTEM_INFO_OPERATION = 'getSystemInformation'
```

Comment:
Can we create Enum like class FTDOperations which will wrap all Operations?

Reply:
Good point, updated.